### ISSUE TITLE
Add 'id' to attrs of 'start_keyword' and 'end_keyword' on listeners v2

### ISSUE THREAD

**Issue by vkruoso** [robotframework/robotframework#2115]
The `start_keyword` and `end_keyword` does not provide the keyword `id` field in the attributes dictionary. As discussed in #1208 this may be added to the version 2 of the listener API.

**Comment by pekkaklarck** [comment]
Yeah, it's a good idea to add ids also to keyword related listener methods. On a quick look PR #2116 looks so good that I don't see any problems getting this into 2.9.1. I'll take a bit closer look at the PR later.

**Comment by pekkaklarck** [comment]
As I already commented PR #2116, keywords don't have enough context information at run time to get correct ids. Changing that is too big and risky task for RF 2.9.1. If you are interested, we can return to this in RF 3.0 that we plan to start developing already next month.

**Comment by vkruoso** [comment]
That is OK. I'm definitely interested to work more on this. Just to be sure, in RF 3.0 there will be a new listener API version that takes real objects as arguments. Will this issue also affect the new API? At a high level view if seems that they complement each other.

**Comment by pekkaklarck** [comment]
Yes, the plan is that RF 3.0 contains listener API v3 that gets real model and/or result objects as arguments (#1208). Adding ids to information passed to listener API v2 makes sense anyway, as long as the needed information is available somewhat easily. Having that information available is actually a requirement for API v3 if we want it contain `start/end_keyword` methods.

**Comment by pekkaklarck** [comment]
The listener v2 API still doesn't add `id` to `start/end_keyword` methods, but this information is available via the listener v3 API that finally got `start/end_keyword` methods in RF 7.0 (#3296). The `id` attribute is dynamic and although generating it is fast, there's still small overhead. I believe having the `id` available only via the more powerful v3 API is enough.
